packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -0,0 +1,562 @@
+      child.parentData = new _ListWheelParentData();
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    _offset.addListener(_hasScrolled);
+  }
+
+  @override
+  void detach() {
+    _offset.removeListener(_hasScrolled);
+    super.detach();
+  }
+
+  @override
+  bool get isRepaintBoundary => true;
+
+  /// Main axis length in the untransformed plane.
+  double get _viewportExtent {
+    assert(hasSize);
+    return size.height;
+  }
+
+  /// Main axis scroll extent in the **scrollable layout coordinates** that puts
+  /// the first item in the center.
+  double get _minScrollExtent {
+    assert(hasSize);
+    return 0.0;
+  }
+
+  /// Main axis scroll extent in the **scrollable layout coordinates** that puts
+  /// the last item in the center.
+  double get _maxScrollExtent {
+    assert(hasSize);
+    if (!(childCount > 0))
+      return 0.0;
+
+    return math.max(0.0, (childCount - 1) * _itemExtent);
+  }
+
+  /// Scroll extent distance in the untransformed plane between the center
+  /// position in the viewport and the top position in the viewport.
+  ///
+  /// It's also the distance in the untransformed plane that children's painting
+  /// is offset by with respect to those children's [BoxParentData.offset].
+  double get _topScrollMarginExtent {
+    assert(hasSize);
+    // Consider adding alignment options other than center.
+    return _minScrollExtent - size.height / 2.0 + _itemExtent / 2.0;
+  }
+
+  /// Transforms a **scrollable layout coordinates**' Y position to the
+  /// **untransformed plane's viewport painting coordinates**' Y position given
+  /// the current scroll offset.
+  double _getUntransformedPaintingCoordinateY(double layoutCoordinateY) {
+    return layoutCoordinateY - _topScrollMarginExtent - offset.pixels;
+  }
+
+  /// Children with offsets larger than this value in the **scrollable layout
+  /// coordinates** can be painted.
+  double get _firstVisibleLayoutOffset {
+    assert(hasSize);
+    if (_renderChildrenOutsideViewport)
+      return double.negativeInfinity;
+    return _minScrollExtent - size.height / 2.0 - _itemExtent / 2.0 + offset.pixels;
+  }
+
+  /// Children with offsets smaller than this value in the **scrollable layout
+  /// coordinates** can be painted.
+  double get _lastVisibleLayoutOffset {
+    assert(hasSize);
+    if (_renderChildrenOutsideViewport)
+      return double.infinity;
+    return _minScrollExtent + size.height / 2.0 + _itemExtent / 2.0 + offset.pixels;
+  }
+
+  /// Given the _diameterRatio, return the largest absolute angle of the item
+  /// at the edge of the portion of the visible cylinder.
+  ///
+  /// For a _diameterRatio of 1 or less than 1 (i.e. the viewport is bigger
+  /// than the cylinder diameter), this value reaches and clips at pi / 2.
+  ///
+  /// When the center of children passes this angle, they are no longer painted
+  /// if renderChildrenOutsideViewport is false.
```

Comment:
use square brackets around the names of other properties, or use // instead of ///

Reply:
Done